financial-services-accelerator/components/org.wso2.financial.services.accelerator.identity.extensions/src/main/java/org/wso2/financial/services/accelerator/identity/extensions/dcr/validators/RequiredParamsValidator.java

```diff
@@ -0,0 +1,189 @@
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.financial.services.accelerator.identity.extensions.dcr.validators;
+
+import com.google.gson.Gson;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.common.model.ServiceProviderProperty;
+import org.wso2.carbon.identity.oauth.dcr.bean.ApplicationRegistrationRequest;
+import org.wso2.carbon.identity.oauth.dcr.bean.ApplicationUpdateRequest;
+import org.wso2.financial.services.accelerator.common.exception.FinancialServicesException;
+import org.wso2.financial.services.accelerator.identity.extensions.internal.IdentityExtensionsDataHolder;
+import org.wso2.financial.services.accelerator.identity.extensions.util.IdentityCommonConstants;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Validator class for validating the required parameters.
+ */
+public class RequiredParamsValidator implements DynamicClientRegistrationValidator {
+
+    private static final Log log = LogFactory.getLog(RequiredParamsValidator.class);
+    private static final Gson gson = new Gson();
+
+    @Override
+    public void validatePost(ApplicationRegistrationRequest applicationRegistrationRequest,
+                             Map<String, Object> ssaParams) throws FinancialServicesException {
+
+        Map<String, Object> requestParameterMap = gson.fromJson(gson.toJson(applicationRegistrationRequest), Map.class);
+        validateRequiredAttributes(requestParameterMap, ssaParams);
+    }
+
+    @Override
+    public void validateGet(Map<String, String> ssaParams) throws FinancialServicesException {
+
+    }
+
+    @Override
+    public void validateUpdate(ApplicationUpdateRequest applicationUpdateRequest, Map<String, Object> ssaParams,
+                               ServiceProviderProperty[] serviceProviderProperties) throws FinancialServicesException {
+
+
+        Map<String, Object> requestParameterMap = gson.fromJson(gson.toJson(applicationUpdateRequest), Map.class);
+        validateRequiredAttributes(requestParameterMap, ssaParams);
+    }
+
+    /**
+     * Validate the required parameters.
+     */
+    private static void validateRequiredAttributes(Map<String, Object> requestParamMap, Map<String, Object> ssaParams)
```

Comment:
Shall we refactor and split this into smaller methods?

Reply:
Fixed in https://github.com/wso2/financial-services-accelerator/pull/327/commits/9bdd95d9a4517faed8c734812dfcd9065d731ce0